### PR TITLE
Deprecate desc_signature['first']

### PR DESCRIPTION
**Pull request by tk0miya**
### Feature or Bugfix
- Refactoring

### Purpose
- Deprecate `desc_signature['first']` attribute
- It never become `True`. So no longer used.

### CODE REVIEW

File: sphinx/domains/c.py [hunk 1]
```diff
@@ -203,7 +203,6 @@ def add_target_and_index(self, name: str, sig: str, signode: desc_signature) ->
         if targetname not in self.state.document.ids:
             signode['names'].append(targetname)
             signode['ids'].append(targetname)
-            signode['first'] = (not self.names)
```
Comment (tk0miya) [inline sphinx/domains/c.py:206]: Before calling `add_target_and_index()`, `self.names` are always filled. So this assigns `False` always.

File: sphinx/writers/html.py [hunk 2]
```diff
@@ -115,10 +115,6 @@ def depart_desc(self, node: Element) -> None:
     def visit_desc_signature(self, node: Element) -> None:
         # the id is set automatically
         self.body.append(self.starttag(node, 'dt'))
-        # anchor for per-desc interactive data
-        if node.parent['objtype'] != 'describe' \
-           and node['ids'] and node['first']:
-            self.body.append('<!--[%s]-->' % node['ids'][0])
```
Comment (tk0miya) [inline sphinx/writers/html.py:121]: It seems the `first` flag has been used to output this comment tag. I don't know why this is needed. Anyway, this is no longer working. So we can drop this feature now.

**Review by shimizukawa — APPROVED** [review]
LGTM!

--

I found first appearance of 'first':
https://github.com/sphinx-doc/sphinx/commits/d60ca8d49d1654968886612883a2affa9dccfaa8/sphinx/directives.py

'first' is functional in the commit like:
https://github.com/sphinx-doc/sphinx/blob/d60ca8d49d1654968886612883a2affa9dccfaa8/sphinx/directives.py#L279-L304

However, no one seems to be in trouble now, so I think it's okay to delete it.

**Comment by tk0miya** [comment]
Thank you for investigation. You're archenologist!

**Comment by tk0miya** [comment]
Note: Merged into 3.x branch manually.

**Pull request merged**